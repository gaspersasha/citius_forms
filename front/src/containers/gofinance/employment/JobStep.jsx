import React from 'react';
import PropTypes from 'prop-types';
import { InputText } from '~components';
import s from './employment.module.sass';

const JobStep = ({ employmentStatus, onJobInfoChange, employer, jobTitle }) => (
  <div className={s.jobFormWrapper}>
    {employmentStatus !== 'selfEmployed' && (
      <div className={s.jobInputWrapper}>
        <InputText
          label="Who is your employer?"
          name="employer"
          subTitle="Please provide your employer below"
          placeholder="Employer"
          required
          onChange={onJobInfoChange}
          status={employer.status}
          value={employer.value}
        />
      </div>
    )}

    <InputText
      label="What is your job title?"
      name="jobTitle"
      subTitle="Please provide your job title below"
      placeholder="Job title"
      required
      onChange={onJobInfoChange}
      status={jobTitle.status}
      value={jobTitle.value}
    />
  </div>
);

JobStep.propTypes = {
  employmentStatus: PropTypes.string.isRequired,
  onJobInfoChange: PropTypes.func.isRequired,
  employer: PropTypes.shape({
    status: PropTypes.string,
    value: PropTypes.string,
  }).isRequired,
  jobTitle: PropTypes.shape({
    status: PropTypes.string,
    value: PropTypes.string,
  }).isRequired,
};

export default JobStep;
