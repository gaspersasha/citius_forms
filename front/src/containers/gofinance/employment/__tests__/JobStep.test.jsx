import React from 'react';
import { INPUT_STATUS } from '~constants';
import JobStep from '../JobStep';

describe('<JobStep />', () => {
  const props = {
    employmentStatus: 'employed',
    onJobInfoChange: jest.fn(),
    employer: {
      value: 'Qwerty',
      status: INPUT_STATUS.VALID,
    },
    jobTitle: {
      value: 'Qwerty',
      status: INPUT_STATUS.VALID,
    },
  };

  const wrapper = shallow(<JobStep {...props} />);

  it('renders without errors', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
