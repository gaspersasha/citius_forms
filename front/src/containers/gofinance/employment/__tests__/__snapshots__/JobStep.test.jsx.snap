// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JobStep /> renders without errors 1`] = `
<div
  className="jobFormWrapper"
>
  <div
    className="jobInputWrapper"
  >
    <InputText
      className=""
      customClasses=""
      errorMessage=""
      label="Who is your employer?"
      name="employer"
      onChange={[MockFunction]}
      placeholder="Employer"
      required={true}
      status="valid"
      subTitle="Please provide your employer below"
      value="Qwerty"
    />
  </div>
  <InputText
    className=""
    customClasses=""
    errorMessage=""
    label="What is your job title?"
    name="jobTitle"
    onChange={[MockFunction]}
    placeholder="Job title"
    required={true}
    status="valid"
    subTitle="Please provide your job title below"
    value="Qwerty"
  />
</div>
`;
